app.js

```diff
@@ -598,39 +651,23 @@ Promise.all([findCompilers(), aws.initConfig(awsProps)])
                 res.set('Content-Type', 'application/xml');
                 res.render('sitemap');
             })
-            .use(sFavicon(`${staticDir}/favicon.ico`))
-            .use('/v', express.static(`${staticDir}/v`, {maxAge: Infinity, index: false}))
-            .use(express.static(staticDir, {maxAge: staticMaxAgeSecs * 1000}));
+            .use(sFavicon(path.join(staticDir, webpackConfig.output.publicPath, 'favicon.ico')));
         if (archivedVersions) {
             // The archived versions directory is used to serve "old" versioned data during updates. It's expected
             // to contain all the SHA-hashed directories from previous versions of Compiler Explorer.
             logger.info("  serving archived versions from", archivedVersions);
-            router.use('/v', express.static(archivedVersions, {maxAge: Infinity, index: false}));
+            webServer.use('/v', express.static(archivedVersions, {maxAge: Infinity, index: false}));
         }
-        router
+        webServer
             .use(bodyParser.json({limit: ceProps('bodyParserLimit', maxUploadSize)}))
             .use(bodyParser.text({limit: ceProps('bodyParserLimit', maxUploadSize), type: () => true}))
             .use(restreamer())
             .use('/source', sourceHandler.handle.bind(sourceHandler))
             .use('/api', apiHandler.handle)
             .use('/g', shortUrlHandler);
-
-        const healthCheck = require('./lib/handlers/health-check');
-        webServer
-            .use(Raven.requestHandler())
-            .set('trust proxy', true)
-            .set('view engine', 'pug')
-            .use('/healthcheck', new healthCheck.HealthCheckHandler().handle) // before morgan so healthchecks aren't logged
-            .use(morgan('combined', {stream: logger.stream}))
-            .use(compression())
-            .use(router)
-            .use(Raven.errorHandler())
-            .on('error', err => logger.error('Caught error:', err, "(in web error handler; continuing)"));
-
-        const webAlias = ceProps("web-alias", "");
```

Comment:
@jaredwy is this now needed? How should this be implemented?

Reply:
I'm more than happy to fix this up after the merge: we're not using it (yet), but once we go outside of docker/nginx we'll need something like this to do beta.